src/cdk-experimental/tabs/tabs.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  booleanAttribute,
+  computed,
+  contentChildren,
+  Directive,
+  ElementRef,
+  inject,
+  input,
+  model,
+} from '@angular/core';
+import {Directionality} from '@angular/cdk/bidi';
+import {TabPattern} from '@angular/cdk-experimental/ui-patterns/tabs/tab';
+import {TablistPattern} from '@angular/cdk-experimental/ui-patterns/tabs/tablist';
+import {TabpanelPattern} from '@angular/cdk-experimental/ui-patterns/tabs/tabpanel';
+import {toSignal} from '@angular/core/rxjs-interop';
+import {_IdGenerator} from '@angular/cdk/a11y';
+
+/**
+ * A Tabs container.
+ *
+ * Represents a set of layered sections of content. The CdkTabs is a container meant to be used with
+ * CdkTablist, CdkTab, and CdkTabpanel as follows:
+ *
+ * ```html
+ * <div cdkTabs>
+ *   <ul cdkTablist>
+ *     <li cdkTab>Tab 1</li>
+ *     <li cdkTab>Tab 2</li>
+ *     <li cdkTab>Tab 3</li>
+ *   </ul>
+ *
+ *   <div cdkTabpanel>Tab content 1</div>
+ *   <div cdkTabpanel>Tab content 2</div>
+ *   <div cdkTabpanel>Tab content 3</div>
+ * </div>
+ * ```
+ */
+@Directive({
+  selector: '[cdkTabs]',
+  exportAs: 'cdkTabs',
+  host: {
+    'class': 'cdk-tabs',
+  },
+})
+export class CdkTabs {
+  /** The CdkTabs nested inside of the container. */
+  private readonly _cdkTabs = contentChildren(CdkTab, {descendants: true});
```

Comment:
Should we start without having `descendants: true`? The problem is that this will also pick up tabs and panels inside of nested tabs, which apparently is somewhat common.